Source/Runtime/MeshPrimitive.cs

```
@@ -280,7 +290,21 @@ public glTFLoader.Schema.MeshPrimitive ConvertToSchema(Runtime.GLTF gltf, List<g
                 bufferViews.Add(bufferView);
                 int bufferviewIndex = bufferViews.Count() - 1;
 
-                glTFLoader.Schema.Accessor accessor = CreateAccessor(bufferviewIndex, 0, glTFLoader.Schema.Accessor.ComponentTypeEnum.UNSIGNED_INT, Indices.Count(), "Indices Accessor", null, null, glTFLoader.Schema.Accessor.TypeEnum.SCALAR, null);
+                var indiceComponentType = glTFLoader.Schema.Accessor.ComponentTypeEnum.UNSIGNED_INT;
+
+                switch (IndiceComponentType)
+                {
+                    case IndiceComponentTypeEnum.UNSIGNED_BYTE:
+                        indiceComponentType = glTFLoader.Schema.Accessor.ComponentTypeEnum.UNSIGNED_BYTE;
+                        break;
+                    case IndiceComponentTypeEnum.UNSIGNED_SHORT:
+                        indiceComponentType = glTFLoader.Schema.Accessor.ComponentTypeEnum.UNSIGNED_SHORT;
+                        break;
+                    default: //Defaults to glTFLoader.Schema.Accessor.ComponentTypeEnum.UNSIGNED_INT;
```

nit: I would prefer switch statements to include all the enum values and have a default that throws. That way if a new enum value is added, this code will fail explicitly.

---

gotcha